Improve docs for `@now/go` (#1204)
zeit/docs: pages/docs/v2/advanced/builders.mdx

<Tag>Status: Alpha</Tag>

> With exported function that has `w http.ResponseWriter, r *http.Request` signature, the Go Builder is used by Now to compile Go Serverless Functions that expose a single HTTP handler, from a `.go` file within an `/api` directory at your project's root.

Reviewer: I don't quite understand this sentence @sophearak @styfle. Does this mean that the Go Builder will only compile files that have one exported function that makes use of these signatures? Can you elaborate?

For advanced usage, such as using private packages with your Go projects, see the [Advanced Go Usage section](#advanced-go-usage).

<Note>

>   The exported function has to have{' '}

Reviewer: ```suggestion
  The exported function needs to include the{' '}
```

  <GenericLink href="https://golang.org/pkg/net/http/#HandlerFunc">
    <InlineCode>HandlerFunc</InlineCode>
  </GenericLink>{' '}

>   signature type. The function name could be any valid Go exported function

Reviewer: ```suggestion
  signature type, but can use any valid Go exported function declaration as the function name.
```

>   declaration.

Reviewer: ```suggestion
```

  using the Go Builder.
</Caption>

> **The entry point of this Builder is a global matching `.go` files** that export a function that implements the `http.HandlerFunc` signature. For more information on using this Builder, [see the Go Builder section](#go).

Reviewer: Just one function? Can they export others? Will it error? etc.
